components/script/window_named_properties.rs

```diff
@@ -0,0 +1,263 @@
+        set_property_descriptor(
+            RustMutableHandle::from_raw(desc),
+            rval.handle(),
+            0,
+            &mut *is_none,
+        );
+    }
+    return true;
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn own_property_keys(
+    cx: *mut JSContext,
+    _proxy: HandleObject,
+    props: MutableHandleIdVector,
+) -> bool {
+    rooted!(in(cx) let mut rooted = SymbolId(GetWellKnownSymbol(cx, SymbolCode::toStringTag)));
+    AppendToIdVector(props, rooted.handle().into());
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn define_property(
+    _cx: *mut JSContext,
+    _proxy: HandleObject,
+    _id: HandleId,
+    _desc: Handle<PropertyDescriptor>,
+    result: *mut ObjectOpResult,
+) -> bool {
+    (*result).code_ = JSErrNum::JSMSG_CANT_DEFINE_WINDOW_NAMED_PROPERTY as usize;
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn delete(
+    _cx: *mut JSContext,
+    _proxy: HandleObject,
+    _id: HandleId,
+    result: *mut ObjectOpResult,
+) -> bool {
+    (*result).code_ = JSErrNum::JSMSG_CANT_DELETE_WINDOW_NAMED_PROPERTY as usize;
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn get_prototype_if_ordinary(
+    _cx: *mut JSContext,
+    proxy: HandleObject,
+    is_ordinary: *mut bool,
+    proto: MutableHandleObject,
+) -> bool {
+    *is_ordinary = true;
+    proto.set(js::jsapi::GetStaticPrototype(proxy.get()));
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn prevent_extensions(
+    _cx: *mut JSContext,
+    _proxy: HandleObject,
+    result: *mut ObjectOpResult,
+) -> bool {
+    (*result).code_ = JSErrNum::JSMSG_CANT_PREVENT_EXTENSIONS as usize;
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn is_extensible(
+    _cx: *mut JSContext,
+    _proxy: HandleObject,
+    extensible: *mut bool,
+) -> bool {
+    *extensible = true;
+    true
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn class_name(_cx: *mut JSContext, _proxy: HandleObject) -> *const libc::c_char {
+    &b"WindowProperties\0" as *const _ as *const _
+}
+
+// Maybe this should be a DOMJSClass. See https://bugzilla.mozilla.org/show_bug.cgi?id=787070
+#[allow(unsafe_code)]
+static CLASS: JSClass = JSClass {
+    name: b"WindowProperties\0" as *const u8 as *const libc::c_char,
+    flags: JSClass_NON_NATIVE |
+        JSCLASS_IS_PROXY |
+        JSCLASS_DELAY_METADATA_BUILDER |
+        ((1 & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT), /* JSCLASS_HAS_RESERVED_SLOTS(1) */
+    cOps: unsafe { &ProxyClassOps },
+    spec: ptr::null(),
+    ext: unsafe { &ProxyClassExtension },
+    oOps: unsafe { &ProxyObjectOps },
+};
+
+#[allow(unsafe_code)]
+pub fn create(
+    cx: SafeJSContext,
+    proto: RustHandleObject,
+    mut properties_obj: RustMutableHandleObject,
+) {
+    unsafe {
+        properties_obj.set(NewProxyObject(
+            *cx,
+            HANDLER.0,
+            UndefinedHandleValue,
+            proto.get(),
+            // TODO: pass proper clasp
```

Comment:
From reading https://searchfox.org/mozilla-central/rev/af78418c4b5f2c8721d1a06486cf4cf0b33e1e8d/dom/base/WindowNamedPropertiesHandler.cpp#244 and https://searchfox.org/mozilla-central/source/js/public/Proxy.h#744-751 I believe we can delete this comment.

Reply:
done in 618ef5744645bb14405d95177bf49aa3b98439dc

Reply:
done in 618ef5744645bb14405d95177bf49aa3b98439dc